File: challenge-004/veesh-goldman/perl5/ch-02.pl
#! /usr/bin/perl

my $usage = <<"EOF";
USAGE:
$0  letters [file]

   letters - a string consisting of the letters to filter by
   file    - the name of a file to check for words. If ommited, checks STDIN

$0 will search through the file for words that can be made from 
the letters in the list argument. Each letter will only be used once.
EOF

die $usage unless @ARGV;

my $list = shift;

while (<>) {
  #remove newline
  chomp;
  #copy the string to check for letters
  my $string = $_;
  #remove any letter in our list from the word
  $string =~ s/$_//i for split //, $list;
  #if there's no word left, then print our original word
  CORE::say unless $string;
}

File: challenge-229/steven-wilson/python/ch-02.py
#!/usr/bin/env python3

from src import two_out_of_three as tt

if __name__ == '__main__':
    print(tt.two_out_of_three([1, 1, 2, 4], [2, 4], [4]))
    print(tt.two_out_of_three([4, 1], [2, 4], [1, 2]))
